### PR TITLE
Feature/mongolog

### PR DESCRIPTION
**Pull request by Kalasouski**

### CODE REVIEW

File: internal/handlers/auth_handlers.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+func (h Handler) registerHandler(ctx *gin.Context) {
+	user := models.RegisterUser{}
+	err := ctx.BindJSON(&user)
+	if err != nil {
+		h.sendErrorResponse(ctx, http.StatusUnprocessableEntity, responses.NewServerRequestError(fmt.Sprint(err)))
+		return
+	}
+	token, err := h.Service.Register(user)
+	if err != nil {
+		h.sendErrorResponse(ctx, usererrors.ErrorToCode(err), responses.NewServerRequestError(fmt.Sprint(err)))
```
Comment (DrGermanius) [inline internal/handlers/auth_handlers.go:24]: what is the purpose of fmt.Sprint() ?

File: internal/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	RequestType string `bson:"request_type"`
+	Err         string
+}
+
+func InitLogger(s *settings.MongoDBSettings) (*MongoLogger, error) {
+	credential := options.Credential{
+		Username: s.User,
+		Password: s.Password,
+	}
+	uri := "mongodb://" + s.Host + ":" + s.Port
```
Comment (DrGermanius) [inline internal/logger/logger.go:29]: better to use Sprintf

File: internal/logger/logger.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	clientOptions := options.Client().ApplyURI(uri).SetAuth(credential)
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatal(err)
```
Comment (DrGermanius) [inline internal/logger/logger.go:33]: why log.Fatal?

Comment (DrGermanius) [reply internal/logger/logger.go:33]: better to return the err

**Review by DrGermanius — CHANGES_REQUESTED** [review]

File: internal/handlers/auth_handlers.go (hunk 1 above)
Comment (DrGermanius) [inline internal/handlers/auth_handlers.go:24]: im still don't understand the purpose of Sprint

File: internal/handlers/middlewares.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		return
+	}
+	claims, err := h.Service.ParseToken(token)
+
+	if err != nil {
+		h.sendErrorResponse(ctx, http.StatusUnauthorized, responses.NewServerUnauthorizedResponse(err.Error()))
+		return
+	}
+	ctx.Set("Claims", claims)
+	ctx.Next()
```
Comment (DrGermanius) [inline internal/handlers/middlewares.go:24]: is that call necessary ?

File: internal/services/service.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+type Token struct {
+	Token string `json:"token"`
+}
+
+func (s Service) Register(user models.RegisterUser) (*Token, error) {
+	_, err := s.Repo.GetUserByPhone(user.Phone)
+	if err == nil {
+		return nil, usererrors.ErrUserExists
+	}
```
Comment (DrGermanius) [inline internal/services/service.go:27]: it's bad solution because what if you have an error during GetUserByPhone call?

**Review by DrGermanius — CHANGES_REQUESTED** [review]

File: internal/handlers/auth_handlers.go (hunk 1 above)
Comment (Kalasouski) [reply internal/handlers/auth_handlers.go:24]: I forgot to delete Sprint in one place

**Review by DrGermanius — APPROVED** [review]
for the following tasks create an another PR

**Pull request merged**